flink-table-store-connector/src/main/java/org/apache/flink/table/store/connector/source/FileStoreSource.java

```diff
@@ -100,6 +100,9 @@ public SplitEnumerator<FileStoreSourceSplit, PendingSplitsCheckpoint> restoreEnu
                     // Only full compaction results will appear on the max level.
                     plan = scan.withLevel(table.options().numLevels() - 1).plan();
                 } else {
+                    if (!isContinuous) {
```

Comment:
For full streaming reading, I think we should enable read compacted too.
In essence, we want to introduce this option to make the read merging more efficient. Full incremental streaming read also requires this capability in the full reading part.